### DISCUSSION TITLE
Only allow users to create suggested languages

### DISCUSSION THREAD

**Discussion by fflorent** [WeblateOrg/weblate#11369] [Q&A] [answered]
We use the hosted.weblate.org to translate documentation written in markdown (and transformed to HTML using mkdocs) that is hosted in Github.

We would like to have the supported languages set in our repository (Github).

The "Add missing language" addon is almost answering our need. We would just like the users to be able to add the suggested languages (the green square below), but not to add any other languages (through the button in red square).

![List of the project languages where is suggested the French language but also the user is given the possibility to add other languages](https://github.com/WeblateOrg/weblate/assets/371705/97c1d475-7142-4678-918d-2e1913339697)

Is a such thing possible?

Thanks in advance

**Comment by fflorent** [comment]
After having checked this conversation: https://github.com/WeblateOrg/weblate/discussions/9703

I might have found an idea: use the API to update every components in the project to set the `language_regex` value to restrict the languages supported. Even the people with the `Manage languages` role should not be able to add other languages than the one that match the regex.

I'll let you know whether this solution works.

**Reply by fflorent** [reply, marked as answer]
Achieved to make something work the way I intended using the API:
https://github.com/gristlabs/grist-help/pull/327

**Reply by nijel** [reply]
Yes, `language_regex` should work for the filtering here, but it is applied when adding the language, so users still see a full list.
